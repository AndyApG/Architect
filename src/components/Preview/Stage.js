import { connect } from 'react-redux';
import { compose } from 'recompose';
import Stage from '../../network-canvas/src/containers/Stage';

const mapStateToProps = (state, props) => {
  return {
    stage: state.protocol.stages[props.stageIndex],
    promptId: state.sessions[state.session].promptIndex,
  };
};

export default compose(
  connect(mapStateToProps),
)(
  Stage,
);
