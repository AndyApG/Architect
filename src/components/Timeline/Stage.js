import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';
import cx from 'classnames';
import { Zoom } from '../../behaviours';
import { getCSSVariableAsString } from '../../utils/CSSVariables';
import timelineImages from '../../images/timeline';

const zoomColors = [getCSSVariableAsString('--light-background'), '#ffffff'];

const getTimelineImage = type =>
  get(timelineImages, type);

const EditStageButton = Zoom(
  ({
    onEditStage,
    type,
    label,
  }) => (
    <div className="timeline-stage__edit-stage">
      <div className="timeline-stage__edit-stage-title">{label || '\u00A0'}</div>
      <div
        className="timeline-stage__screen"
        role="button"
        onClick={onEditStage}
        tabIndex="0"
      >
        <div className="timeline-stage__screen-preview">
          {
            getTimelineImage(type) &&
            <img
              src={getTimelineImage(type)}
              alt={`${type} Interface`}
              title={`${type} Interface`}
            />
          }
          {
            !getTimelineImage(type) &&
            `${type} Interface`
          }
        </div>
      </div>
    </div>

  ),
);

const ConfigureSkipLogicButton = Zoom(
  ({
    onEditSkipLogic,
  }) => (
    <a
      className="timeline-stage__control"
      onClick={onEditSkipLogic}
    >
      <div className="timeline-stage__control-icon">&#9881;</div>
      Configure skip logic
    </a>
  ),
);

class TimelineStage extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    label: PropTypes.string,
    type: PropTypes.string.isRequired,
    onEditStage: PropTypes.func.isRequired,
    onDeleteStage: PropTypes.func.isRequired,
    onEditSkipLogic: PropTypes.func.isRequired,
    onInsertStage: PropTypes.func.isRequired,
  };

  static defaultProps = {
    label: '',
    className: '',
  };

  render() {
    const {
      className,
      onEditStage,
      onDeleteStage,
      onEditSkipLogic,
      onInsertStage,
      type,
      label,
      ...rest
    } = this.props;

    const componentClasses = cx(
      'timeline-stage',
      className,
    );

    return (
      <div className={componentClasses} {...rest}>
        <div
          className="timeline-stage__notch"
          onClick={onEditStage}
        />
        <EditStageButton
          onEditStage={onEditStage}
          type={type}
          label={label}
          zoomColors={zoomColors}
        />
        <div className="timeline-stage__controls">
          <a
            className="timeline-stage__control"
            onClick={() => onInsertStage(0)}
          >
            <div className="timeline-stage__control-icon">↑</div>
            Add screen before
          </a>
          <ConfigureSkipLogicButton
            onEditSkipLogic={onEditSkipLogic}
            zoomColors={['#ffffff', '#f2b700']}
          />
          <a
            className="timeline-stage__control"
            onClick={onDeleteStage}
          >
            <div className="timeline-stage__control-icon">✕</div>
            Delete stage
          </a>
          <a
            className="timeline-stage__control"
            onClick={() => onInsertStage(1)}
          >
            <div className="timeline-stage__control-icon">↓</div>
            Add screen after
          </a>
        </div>
      </div>
    );
  }
}

export default TimelineStage;
