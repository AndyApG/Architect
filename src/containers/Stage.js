/* eslint-disable */

import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

class Stage extends PureComponent {
  static propTypes = {
    stages: PropTypes.array.isRequired,
  };

  render() {
    return (
      <div className="stage">
        Stage
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    stages: state.stages,
  };
}

function mapDispatchToProps() {
  return {
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Stage);
