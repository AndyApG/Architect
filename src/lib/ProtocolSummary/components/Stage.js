import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import SummaryContext from './SummaryContext';

const stageVariables = (index) => (stageId) => index
  .reduce(
    (memo, variable) => {
      if (!variable.stages.includes(stageId)) { return memo; }
      return [...memo, variable.name];
    },
    [],
  )
  .join(', ');

const Stage = ({
  stageNumber,
  type,
  label,
  id,
  configuration,
}) => {
  const {
    index,
  } = useContext(SummaryContext);

  return (
    <div>
      <div>
        <h1>
          {stageNumber}
          {'. '}
          {label}
        </h1>

        <div>
          {type}
        </div>

        <div>
          <h2>Variables</h2>
          { stageVariables(index)(id) }
        </div>
      </div>

      { configuration.interviewScript && (
        <div>
          <h2>Script</h2>
          {configuration.interviewScript}
        </div>
      )}

      { configuration.prompts && (
        <div>
          <h2>Prompts</h2>
          <ol>
            {configuration.prompts.map((prompt) => (
              <li key={prompt.id}>{prompt.text}</li>
            ))}
          </ol>
        </div>
      )}
    </div>
  );
};

Stage.propTypes = {
  stageNumber: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  configuration: PropTypes.object.isRequired,
};

export default Stage;
