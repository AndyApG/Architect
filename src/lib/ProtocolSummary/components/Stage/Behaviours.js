import React from 'react';
import PropTypes from 'prop-types';
import MiniTable from '../MiniTable';
import { renderValue } from '../helpers';

const Behaviours = ({ behaviours }) => {
  if (!behaviours) { return null; }

  return (
    <div className="protocol-summary-stage__behaviours">
      <h2>Behaviours</h2>

      <div className="protocol-summary-stage__behaviours-content">
        <MiniTable
          rows={[
            [
              <strong>Repositioning enabled</strong>,
              renderValue(behaviours.allowRepositioning),
            ],
            [
              <strong>Freedraw enabled</strong>,
              renderValue(behaviours.freeDraw),
            ],
          ]}
        />
      </div>
    </div>
  );
};

Behaviours.propTypes = {
  behaviours: PropTypes.shape({
    allowRepositioning: PropTypes.bool,
    freeDraw: PropTypes.bool,
  }),
};

Behaviours.defaultProps = {
  behaviours: null,
};

export default Behaviours;
