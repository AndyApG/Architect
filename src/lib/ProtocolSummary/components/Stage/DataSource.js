import React from 'react';
import PropTypes from 'prop-types';
import AssetBadge from '../AssetBadge';

const DataSource = ({ dataSource }) => {
  if (!dataSource) { return null; }

  return (
    <div className="protocol-summary-stage__data-source">
      <div className="protocol-summary-stage__data-source-content">
        <h2 className="section-heading">DataSource</h2>
        <AssetBadge id={dataSource} link />
      </div>
    </div>
  );
};

DataSource.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  dataSource: PropTypes.object,
};

DataSource.defaultProps = {
  dataSource: null,
};

export default DataSource;
