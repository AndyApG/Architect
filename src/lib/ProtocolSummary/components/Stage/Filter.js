import React from 'react';
import Rules from '../Rules';

const Filter = ({ filter }) => {
  if (!filter) { return null; }

  console.log({ filter });

  return (
    <div className="protocol-summary-stage__filter">
      <Rules filter={filter} />
    </div>
  );
};

export default Filter;
