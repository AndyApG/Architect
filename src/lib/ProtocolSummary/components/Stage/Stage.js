import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import EntityBadge from '../EntityBadge';
import SummaryContext from '../SummaryContext';
import DualLink from '../DualLink';
import Prompts from './Prompts';
import InterviewScript from './InterviewScript';

const variablesOnStage = (index) => (stageId) => index
  .reduce(
    (memo, variable) => {
      if (!variable.stages.includes(stageId)) { return memo; }
      return [...memo, [variable.id, variable.name]];
    },
    [],
  );

const Stage = ({
  configuration,
  id,
  label,
  stageNumber,
  type,
}) => {
  const {
    index,
  } = useContext(SummaryContext);

  const stageVariables = variablesOnStage(index)(id);
  const { subject } = configuration;

  return (
    <div className="protocol-summary-stage" id={`stage-${id}`}>
      <div className="protocol-summary-stage__heading">
        <h1>
          {stageNumber}
          {'. '}
          {label}
        </h1>

        <table className="protocol-summary-stage__meta">
          <tbody>
            <tr>
              <th>Type</th>
              <td>{type}</td>
            </tr>
            { configuration.subject && (
              <tr>
                <th className="protocol-summary-stage__meta-subject-title">Subject</th>
                <td>
                  <EntityBadge type={subject.type} entity={subject.entity} link small />
                </td>
              </tr>
            )}
            {stageVariables.length > 0 && (
              <tr>
                <th>Variables</th>
                <td>
                  { stageVariables.map(([variableId, variable]) => (
                    <>
                      <DualLink to={`#variable-${variableId}`} key={variableId}>{variable}</DualLink>
                      <br key={`br-${variableId}`} />
                    </>
                  )) }
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>

      <div className="protocol-summary-stage__content">
        <Prompts prompts={configuration.prompts} />
        <InterviewScript interviewScript={configuration.interviewScript} />
      </div>
    </div>
  );
};

Stage.propTypes = {
  id: PropTypes.string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  configuration: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  stageNumber: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
};

export default Stage;
