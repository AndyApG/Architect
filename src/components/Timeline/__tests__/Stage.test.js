/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Stage } from '../Stage';

jest.mock('../../../ui/utils/CSSVariables');

const mockProps = {
  type: 'Foo',
  onEditStage: () => {},
  onDeleteStage: () => {},
  onEditSkipLogic: () => {},
  onInsertStage: () => {},
};

describe('<Stage />', () => {
  it('can render', () => {
    const component = shallow(<Stage {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
