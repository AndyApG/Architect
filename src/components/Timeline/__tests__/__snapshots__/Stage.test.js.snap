// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineStage
    className=""
    label=""
    onDeleteStage={[Function]}
    onEditSkipLogic={[Function]}
    onEditStage={[Function]}
    onInsertStage={[Function]}
    type="Foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="timeline-stage__notch"
          onClick={[Function]}
        />,
        <getContext(Draft(Component))
          label=""
          onEditStage={[Function]}
          type="Foo"
          zoomColors={
            Array [
              "#227733",
              "#ffffff",
            ]
          }
        />,
        <div
          className="timeline-stage__controls"
        >
          <a
            className="timeline-stage__control"
            onClick={[Function]}
          >
            <div
              className="timeline-stage__control-icon"
            >
              ↑
            </div>
            Add screen before
          </a>
          <getContext(Draft(Component))
            onEditSkipLogic={[Function]}
            zoomColors={
              Array [
                "#ffffff",
                "#f2b700",
              ]
            }
          />
          <a
            className="timeline-stage__control"
            onClick={[Function]}
          >
            <div
              className="timeline-stage__control-icon"
            >
              ✕
            </div>
            Delete stage
          </a>
          <a
            className="timeline-stage__control"
            onClick={[Function]}
          >
            <div
              className="timeline-stage__control-icon"
            >
              ↓
            </div>
            Add screen after
          </a>
        </div>,
      ],
      "className": "timeline-stage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "timeline-stage__notch",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "",
          "onEditStage": [Function],
          "type": "Foo",
          "zoomColors": Array [
            "#227733",
            "#ffffff",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ↑
              </div>
              Add screen before
            </a>,
            <getContext(Draft(Component))
              onEditSkipLogic={[Function]}
              zoomColors={
                Array [
                  "#ffffff",
                  "#f2b700",
                ]
              }
            />,
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ✕
              </div>
              Delete stage
            </a>,
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ↓
              </div>
              Add screen after
            </a>,
          ],
          "className": "timeline-stage__controls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="timeline-stage__control-icon"
                >
                  ↑
                </div>,
                "Add screen before",
              ],
              "className": "timeline-stage__control",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "↑",
                  "className": "timeline-stage__control-icon",
                },
                "ref": null,
                "rendered": "↑",
                "type": "div",
              },
              "Add screen before",
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onEditSkipLogic": [Function],
              "zoomColors": Array [
                "#ffffff",
                "#f2b700",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="timeline-stage__control-icon"
                >
                  ✕
                </div>,
                "Delete stage",
              ],
              "className": "timeline-stage__control",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "✕",
                  "className": "timeline-stage__control-icon",
                },
                "ref": null,
                "rendered": "✕",
                "type": "div",
              },
              "Delete stage",
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="timeline-stage__control-icon"
                >
                  ↓
                </div>,
                "Add screen after",
              ],
              "className": "timeline-stage__control",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "↓",
                  "className": "timeline-stage__control-icon",
                },
                "ref": null,
                "rendered": "↓",
                "type": "div",
              },
              "Add screen after",
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="timeline-stage__notch"
            onClick={[Function]}
          />,
          <getContext(Draft(Component))
            label=""
            onEditStage={[Function]}
            type="Foo"
            zoomColors={
              Array [
                "#227733",
                "#ffffff",
              ]
            }
          />,
          <div
            className="timeline-stage__controls"
          >
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ↑
              </div>
              Add screen before
            </a>
            <getContext(Draft(Component))
              onEditSkipLogic={[Function]}
              zoomColors={
                Array [
                  "#ffffff",
                  "#f2b700",
                ]
              }
            />
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ✕
              </div>
              Delete stage
            </a>
            <a
              className="timeline-stage__control"
              onClick={[Function]}
            >
              <div
                className="timeline-stage__control-icon"
              >
                ↓
              </div>
              Add screen after
            </a>
          </div>,
        ],
        "className": "timeline-stage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "timeline-stage__notch",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "",
            "onEditStage": [Function],
            "type": "Foo",
            "zoomColors": Array [
              "#227733",
              "#ffffff",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="timeline-stage__control"
                onClick={[Function]}
              >
                <div
                  className="timeline-stage__control-icon"
                >
                  ↑
                </div>
                Add screen before
              </a>,
              <getContext(Draft(Component))
                onEditSkipLogic={[Function]}
                zoomColors={
                  Array [
                    "#ffffff",
                    "#f2b700",
                  ]
                }
              />,
              <a
                className="timeline-stage__control"
                onClick={[Function]}
              >
                <div
                  className="timeline-stage__control-icon"
                >
                  ✕
                </div>
                Delete stage
              </a>,
              <a
                className="timeline-stage__control"
                onClick={[Function]}
              >
                <div
                  className="timeline-stage__control-icon"
                >
                  ↓
                </div>
                Add screen after
              </a>,
            ],
            "className": "timeline-stage__controls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="timeline-stage__control-icon"
                  >
                    ↑
                  </div>,
                  "Add screen before",
                ],
                "className": "timeline-stage__control",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "↑",
                    "className": "timeline-stage__control-icon",
                  },
                  "ref": null,
                  "rendered": "↑",
                  "type": "div",
                },
                "Add screen before",
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onEditSkipLogic": [Function],
                "zoomColors": Array [
                  "#ffffff",
                  "#f2b700",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="timeline-stage__control-icon"
                  >
                    ✕
                  </div>,
                  "Delete stage",
                ],
                "className": "timeline-stage__control",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "✕",
                    "className": "timeline-stage__control-icon",
                  },
                  "ref": null,
                  "rendered": "✕",
                  "type": "div",
                },
                "Delete stage",
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="timeline-stage__control-icon"
                  >
                    ↓
                  </div>,
                  "Add screen after",
                ],
                "className": "timeline-stage__control",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "↓",
                    "className": "timeline-stage__control-icon",
                  },
                  "ref": null,
                  "rendered": "↓",
                  "type": "div",
                },
                "Add screen after",
              ],
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
